pandas/_libs/lib.pyx

```diff
@@ -1315,6 +1315,14 @@ cdef class Seen:
             or self.numeric_ or self.nan_ or self.null_ or self.object_
         )
 
+    @property
+    def is_bool_or_na(self):
```

Comment:
i think both this and is_bool above should check that `self.bool_` is True.  Not really needed for correctness given the way it is used, but future-proofing against mistakes

Reply:
could also rewrite is_bool as `self.is_bool_or_na and not self.null_`

Reply:
good point, this looks better. Changed